src/main/java/pl/javastart/task/vehicle/Truck.java

```diff
@@ -0,0 +1,53 @@
+package pl.javastart.task.vehicle;
+
+public class Truck extends Car {
+
+    private final String id = "Truck";
+
+    private static final double WEIGHT_EXTRA_FUEL = .5;
+
+    private static final double AIR_CON_CONSUMPTION = 1.6;
+    private double loadWeigth;
+
+    public Truck(String name, double tankCapacity, double averageFuelConsumption, boolean aiconOn, double loadWeigth) {
+        super(name, tankCapacity, averageFuelConsumption, aiconOn);
+        this.loadWeigth = loadWeigth;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public double getLoadWeigth() {
+        return loadWeigth;
+    }
+
+    public void setLoadWeigth(double loadWeigth) {
+        this.loadWeigth = loadWeigth;
+    }
+
+    @Override
+    public String info() {
+        return super.info() + ", Waga ładunku: " + loadWeigth;
+    }
+
+    @Override
+    public double range() {
+        if (loadWeigth == 0) {
+            return tankDividedByFuelConsumption() * BY_100_KM;
+        }
+        return  (tankDividedByFuelConsumption() + weightFuelConsumption()) * BY_100_KM;
```

Comment:
Ten if nie jest potrzebny. Jeżeli. ładunek będzie 0, to metoda `weightFuelConsumption()` zwróci 0 i też będzie ok